src/monopoly.js

```diff
@@ -1,48 +1,158 @@
+    this.cleanBoard();
 
-    findFirstPlayer();
+    this.findFirstPlayer();
 
-    while(PlayerList.length > 1){
-      runTurn(PlayerList.current);
-      PlayerList.Current = PlayerList.Current.Next;//or next index or w.e. I use
+    while(this.PlayerArray.length > 1){
+      if(this.PlayerArray[this.currentPlayer].money < 0){//test
+        this.PlayerArray.splice(this.currentPlayer, 1);
+        continue;
+      }
+      this.runTurn(this.PlayerArray[this.currentPlayer]);
+      this.currentPlayer++;//or next index or w.e. I use
+      if(this.PlayerArray.length >= this.currentPlayer){
+        this.currentPlayer = 0;
+      }
     }
 
-    //add wincount;
+    //TODO add wincount;
   }
 
-  runTurn(){
-    
+  runTurn(ActivePlayer){
+    const thisGame = this;//because 'this' breaks with the event
+    console.log('runTurn for ' + ActivePlayer.person + '. Player position: ' + ActivePlayer.position);//test
+
+    this.Emitter.once('rollDice', function(){
+      thisGame.toggleListenersOff();
+
+      let diceArray = thisGame.rollDice();
+      console.log('dice: ' + diceArray.reduce(( acc, cur ) => acc + cur, 0));//test
+      ActivePlayer.movePlayer(diceArray, thisGame.PropertyArray.length);
+      console.log('new position ' + ActivePlayer.position);//test
+      ActivePlayer.money -= diceArray.reduce(( acc, cur ) => acc + cur, 0) * 10;//test
+//      thisGame.landOnFunction(ActivePlayer);TODO
+
+      if(diceArray.every(element => element == diceArray[0])){//probably too much.  allows for more than 2 dice
+        if(++ActivePlayer.doubles >= 3){
+          ActivePlayer.doubles = 0;
+          thisGame.goToJail(ActivePlayer);
+          return this;
+        }
+        else {
+          thisGame.runTurn(ActivePlayer);
+          return this;
+        }
+      }
+      ActivePlayer.doubles = 0;
+
+      return this;
+    });
+
+    this.toggleListenersOn();
+    console.log('Toggling Listers\n promptRoll for ' + ActivePlayer.person);//test
+    this.Emitter.emit('promptRoll', ActivePlayer);
   }
 
+  boardState(){
+    const CurrentBoardState = new BoardState(this.PropertyArray, this.PlayerArray, this.currentPlayer);
+//    CurrentBoardState.PlayerArray.forEach(function(thisPlayer){
+//      thisPlayer.socketID = null;//TODO this won't work, need to clone or just create new object....just a general idea.
+//      thisPlayer.registeredID = null;
+//    });
+
+    return JSON.stringify(CurrentBoardState, null, 2);
+  }//returns JSON boardState
+
   cleanBoard(){
-    PropertyList = new PropertyList;
-    ChanceList = new ChanceList;
-    CommChestList = new CommChestList;
+    this.PropertyArray = JSON.parse(fs.readFileSync('properties.JSON'));
+//    this.ChanceList = JSON.parse(fs.readFileSync('chance.JSON'));
+//    this.CommChestList = JSON.parse(fs.readFileSync('communityChest.JSON'));
+
+    this.resetPlayers();
+
+    return this;
+  }
 
-    resetPlayers();
+  findFirstPlayer(){//TODO what if ties occur
+    let highRoll = 0;
```

Comment:
Is it important to you that the random-player-chooser-majig simulates what players do in the game?  I.e. rolling dice to see who goes first then doing roll-offs. 

If not, I would just take the list of players, scramble them, and call that the order.  
https://www.npmjs.com/package/random-js

Has a shuffle method.  You can also use it to generate your random ids.  just a thought.